posthog/api/routing.py

```diff
@@ -285,19 +339,8 @@ def _filter_queryset_by_parents_lookups(self, queryset):
             return queryset
 
     @cached_property
-    def parents_query_dict(self) -> dict[str, Any]:
```

Comment:
The only change to `parents_query_dict()` is that "map every lookup-prefixed item in self.kwargs" is now partly pulled out into `parent_query_kwargs()`. This way `... in self.parent_query_kwargs` checks initiate DB queries